### PR TITLE
One CI workflow to rule them all

### PR DESCRIPTION
**Pull request by ajsantander**
Also removes the branch naming filters, so this one workflow to rule them all is always triggered.

### CODE REVIEW

**Review by justinjmoses — APPROVED** [review]

**Comment by justinjmoses** [comment]
That's a weird failure - looks like it was killed? `Error: Invalid JSON RPC response: ""` used to happen from time to time, that why the test publish script has retries on it...

**Review by justinjmoses — APPROVED** [review]

File: hardhat/tasks/task-compile.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ task('compile')
 		if (taskArguments.useOvm) {
 			console.log(gray('Compiling with OVM Solidity compiler...'));
 
-			require('@eth-optimism/plugins/hardhat/compiler/0.5.16');
```
Comment (K-Ho) [inline hardhat/tasks/task-compile.js:28]: does this work? I think you may have to keep it at `@eth-optimism/plugins/hardhat/compiler/0.5.16`

Comment (K-Ho) [reply hardhat/tasks/task-compile.js:28]: nevermind!

Comment (ajsantander) [reply hardhat/tasks/task-compile.js:28]: Yes, according to the docs!
https://hardhat.org/plugins/eth-optimism-plugins-hardhat-compiler.html

**Review by justinjmoses — APPROVED** [review]
:boom:

**Comment by ajsantander** [comment]
> That's a weird failure - looks like it was killed? `Error: Invalid JSON RPC response: ""` used to happen from time to time, that why the test publish script has retries on it...

This turned out to be the CI running out of memory, but not reporting the error properly because of stdout and stderr re-routing done in the tests.

**Pull request merged**